.github/workflows/deploy.yml

```diff
@@ -0,0 +1,39 @@
+name: Deploy to GitHub Pages
+
+on:
+  # Triggers the workflow on push or pull request events but
+  # only for the default branch.
+  push:
+    branches:
+      - master
+    paths:
+      - '.github/workflows/deploy.yml'
+      - 'book.toml'
+      - 'guide/**'
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+
+jobs:
+  fetch:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Clone repository
+        uses: actions/checkout@v4
+      - name: Install Rust packages
+        run: |
+          curl --proto '=https' --tlsv1.2 -sSf https://rossmacarthur.github.io/install/crate.sh | bash -s -- --repo "rust-lang/mdBook" --bin mdbook --to ~/.cargo/bin
```

Comment:
This seems like a very surprising way to install mdBook.

The user guide includes a section on [Running `mdbook` in Continuous Integration](https://rust-lang.github.io/mdBook/continuous-integration.html), which suggests two different methods, in case we're not happy with just
```sh
cargo install mdbook
```
which should also "just work".

A decent indication that the recommended methods work is that they're used to build the [rust-lang docs site](https://github.com/rust-lang/cargo/blob/7bb7b539558dc88bea44cee4168b6269bf8177b0/.github/workflows/main.yml#L244-L248).

Reply:
This is copied from our existing repositories, it looks like it became obsolete. As a disclaimer, this code came from someone very familiar with Rust and, at the time, the only alternative was compiling from source.